libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt

```diff
@@ -2,21 +2,88 @@ package com.mapbox.navigation.route.onboard
 
 import com.mapbox.annotation.navigation.module.MapboxNavigationModule
 import com.mapbox.annotation.navigation.module.MapboxNavigationModuleType
+import com.mapbox.navigation.base.logger.Logger
+import com.mapbox.navigation.base.route.RouteUrl
 import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.base.route.model.Route
 import com.mapbox.navigation.base.route.model.RouteOptionsNavigation
 import com.mapbox.navigation.navigator.MapboxNativeNavigator
+import com.mapbox.navigation.navigator.MapboxNativeNavigatorImpl
+import com.mapbox.navigation.route.onboard.model.Config
+import com.mapbox.navigation.route.onboard.model.OfflineError
+import com.mapbox.navigation.route.onboard.model.mapToRouteConfig
+import com.mapbox.navigation.route.onboard.task.OfflineRouteRetrievalTask
+import com.mapbox.navigation.utils.exceptions.NavigationException
+import java.io.File
 
 @MapboxNavigationModule(MapboxNavigationModuleType.OnboardRouter, skipConfiguration = true)
-class MapboxOnboardRouter(private val navigator: MapboxNativeNavigator) : Router {
+class MapboxOnboardRouter : Router {
+
+    companion object {
+        private const val TILES_DIR_NAME = "tiles"
+    }
+
+    private val navigatorNative: MapboxNativeNavigator
+    private val config: Config
+    private val logger: Logger?
+
+    /**
+     * Creates an offline router which uses the specified offline path for storing and retrieving
+     * data.
+     *
+     * @param config offline config
+     */
+    constructor(config: Config, logger: Logger?) {
+        val tileDir = File(config.tilePath, TILES_DIR_NAME)
+        if (!tileDir.exists()) {
+            tileDir.mkdirs()
+        }
+
+        this.navigatorNative = MapboxNativeNavigatorImpl
+        this.config = config
+        this.logger = logger
+        MapboxNativeNavigatorImpl.configureRouter(config.mapToRouteConfig())
+    }
+
+    // Package private for testing purposes
+    internal constructor(
+        navigator: MapboxNativeNavigator,
+        config: Config
+    ) {
+        this.navigatorNative = navigator
+        this.config = config
+        this.logger = null
+    }
 
     override fun getRoute(
         routeOptions: RouteOptionsNavigation,
         callback: Router.Callback
-    ) = Unit
+    ) {
+        val offlineRouter = OfflineRoute.builder(
```

Comment:
As mentioned somewhere else, `RouteOptionsNavigation` extension could expose all offline route options outside of the URL that could be set on the builder. This can be done as a tailwork.

Reply:
yep, I get your [point](https://github.com/mapbox/mapbox-navigation-android/pull/2342#discussion_r364215798) but lets merge this PR and after improve implementation